src/Compilers/VisualBasic/Test/Semantic/Semantics/NameOfTests.vb

```diff
@@ -3637,5 +3637,32 @@ BC30469: Reference to a non-shared member requires an object reference.
                    ~~~~~~~~~~~~~~~~~~~~~~~~
 ]]></expected>)
         End Sub
+
+        <Fact, WorkItem(23019, "https://github.com/dotnet/roslyn/issues/23019")>
+        Public Sub NameOfInAsync()
+            Dim compilationDef =
+<compilation>
+    <file name="a.vb">
+Imports System
+Imports System.Threading.Tasks
+
+Module Module1
+    Sub Main()
+        M().GetAwaiter().GetResult()
+    End Sub
+    Async Function M() As Task
+        Console.WriteLine(NameOf(M))
+        Await Task.Delay(0)
+    End Function
+End Module
+    </file>
+</compilation>
+
+            Dim comp = CreateCompilationWithMscorlib40AndVBRuntimeAndReferences(compilationDef, {MscorlibRef_v4_0_30316_17626, MsvbRef_v4_0_30319_17929}, TestOptions.DebugExe)
```

Comment:
>{MscorlibRef_v4_0_30316_17626, MsvbRef_v4_0_30319_17929} [](start = 96, length = 56)

This looks suspicious. Why is this needed?

Reply:
I had copy/pasted this from https://github.com/dotnet/roslyn/blob/master/src/Compilers/VisualBasic/Test/Semantic/Semantics/AsyncAwait.vb tests.

Reply:
> I had copy/pasted this from [https://github.com/dotnet/roslyn/blob/master/src/Compilers/VisualBasic/Test/Semantic/Semantics/AsyncAwait.vb](https://github.com/dotnet/roslyn/blob/master/src/Compilers/VisualBasic/Test/Semantic/Semantics/AsyncAwait.vb) tests.

I can believe this, at the time when the tests were created, that probably was the way to go. It is no longer necessary to do now.

---
In reply to: [451269169](https://github.com/dotnet/roslyn/pull/45738#discussion_r451269169) [](ancestors = 451269169)